clients/cli/src/prover_runtime.rs

```diff
@@ -382,13 +382,13 @@ pub fn start_workers(
     let mut handles = Vec::with_capacity(num_workers);
 
     for worker_id in 0..num_workers {
-        let (task_sender, mut task_receiver) = tokio::sync::mpsc::channel::<Task>(8);
+        let (task_sender, mut task_receiver) = mpsc::channel::<Task>(8);
+        // Clone senders and receivers for each worker.
         let prover_event_sender = event_sender.clone();
         let results_sender = results_sender.clone();
-        let mut shutdown = shutdown.resubscribe(); // Clone the receiver for each worker
+        let mut shutdown = shutdown.resubscribe();
         let handle = tokio::spawn(async move {
-            while let Some(task) = task_receiver.recv().await {
-                // Check for shutdown signal
+            loop {
                 tokio::select! {
```

Comment:
Previously, checking for the shutdown signal only happened after a task is received, which may give unpredictable behavior. This PR moves the check up to the same level as receiving tasks so that shutdown is handled immediately